x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetStackTracesAction.java

```diff
@@ -448,36 +459,47 @@ private void retrieveStackTraces(
         if (submitTask.notifyIfCancelled(submitListener)) {
             return;
         }
-        List<String> eventIds = new ArrayList<>(responseBuilder.getStackTraceEvents().keySet());
+        List<String> stacktraceIds = responseBuilder.getStackTraceEvents()
```

Comment:
We're iterating through all stack trace events twice, creating an intermediary set and then end up with a (sorted) list. IMHO we could achieve the same by simply iterating over the stack trace events once in a for(each) loop and using a TreeSet (which is sorted), i.e. something along the lines of:


```java
Set<String> stacktraceIds = new TreeSet<>();
Set<String> hostIds = new TreeSet<>();
for (TraceEventID id : responseBuilder.getStackTraceEvents().keySet()) {
    stacktraceIds.add(id.stacktraceID());
    hostIds.add(id.hostID());
}
```

It's less code to read and way more straightforward. Due to slicing below, we need to wrap this again in a list though but even then this code is simpler to understand IMHO.

Reply:
:+1: I also prefer the old style, not sure why I used the streaming approach (maybe I was playing around with it and FORGOT about it once it worked 🤔).